source/scripting_v3/GTA/Entities/Peds/NavMeshBlockingObject.cs

```diff
@@ -0,0 +1,123 @@
+﻿//
+// Copyright (C) 2015 crosire & contributors
+// License: https://github.com/crosire/scripthookvdotnet#license
+//
+
+using GTA.Math;
+using GTA.Native;
+
+namespace GTA
+{
+	/// <summary>
+	/// Represents a blocking object that blocks <see cref="Ped"/>s using navigation mesh paths in the area the object covers when new <see cref="Ped"/> tasks that use navigation meshes (e.g. <see cref="TaskInvoker.WanderAround()"/>) start for them.
+	/// Does not create any dynamic objects or <see cref="Prop"/>s.
+	/// </summary>
+	/// <remarks>If SHVDN runtime stops working, all the <see cref="NavMeshBlockingObject"/> will get removed from the game.</remarks>
+	public sealed class NavMeshBlockingObject : PoolObject, INativeValue
+	{
+		internal NavMeshBlockingObject(int handle) : base(handle)
+		{
+		}
+
+		/// <summary>
+		/// Removes this <see cref="NavMeshBlockingObject"/>.
+		/// </summary>
+		public override void Delete()
+		{
+			Function.Call(Hash.REMOVE_NAVMESH_BLOCKING_OBJECT, Handle);
+		}
+
+		/// <summary>
+		/// Determines if this <see cref="NavMeshBlockingObject"/> exists.
+		/// </summary>
+		/// <returns><see langword="true" /> if this <see cref="NavMeshBlockingObject"/> exists; otherwise, <see langword="false"/></returns>.
+		public override bool Exists()
+		{
+			return Function.Call<bool>(Hash.DOES_NAVMESH_BLOCKING_OBJECT_EXIST, Handle);
+		}
+
+		const float DEG_2_RAD = (float)(System.Math.PI / 180);
+
+		/// <summary>
+		/// Creates a new <see cref="NavMeshBlockingObject"/>. 
+		/// </summary>
+		/// <param name="position">The origin position.</param>
+		/// <param name="size">The size.</param>
+		/// <param name="headingDegrees">The heading in degrees.</param>
+		/// <param name="flags">The flags that specify what types of paths the blocking object will block new <see cref="Ped"/> tasks that use navigation meshes (e.g. <see cref="TaskInvoker.WanderAround()"/>) from using.</param>
+		/// <returns>The new <see cref="NavMeshBlockingObject"/> if successfully created; otherwise, <see langword="null"/>.</returns>
+		/// <remarks>The new <see cref="NavMeshBlockingObject"/> won't block existing <see cref="Ped"/> tasks that use navigation mesh paths in the area the blocking object covers from using.</remarks>
+		public static NavMeshBlockingObject Create(Vector3 position, Vector3 size, float headingDegrees, NavMeshBlockingObjectFlags flags = NavMeshBlockingObjectFlags.AllPaths)
```

Comment:
This is breaking design with the other object types a bit (which are all created with a `World.CreateObject` method instead of `Object.Create`.